### PR TITLE
Deprecate CurrentTimeUTC

### PR DESCRIPTION
**Pull request by bertm**
Follow-up on discussion in #866

### CODE REVIEW

File: src/freenet/support/CurrentTimeUTC.java [hunk 1]
```diff
@@ -29,30 +40,45 @@ public static Date get() {
...
 	 */
+	@Deprecated
 	public static long getInMillis() {
 		return System.currentTimeMillis();
 	}
 
+	/**
+	 * Get the current calendar year.
+	 * @deprecated use {@link java.time.LocalDate#now(java.time.ZoneId) LocalDate.now(ZoneOffset.UTC)}
+	 */
```
Comment (ArneBab) [inline src/freenet/support/CurrentTimeUTC.java:55]: this replacement does not return the year.

File: src/freenet/support/CurrentTimeUTC.java [hunk 2]
```diff
@@ -29,30 +40,45 @@ public static Date get() {
...
 		synchronized(mCalendar) {
 			mCalendar.setTimeInMillis(System.currentTimeMillis());
 			return mCalendar.get(Calendar.YEAR);
 		}
 	}
 
+	/**
+	 * Get the zero-indexed current calendar month.
+	 * @return the zero-indexed month number, where 0 indicates January and 11 indicates December
+	 * @deprecated use {@link java.time.LocalDate#now(java.time.ZoneId) LocalDate.now(ZoneOffset.UTC)}
```
Comment (ArneBab) [inline src/freenet/support/CurrentTimeUTC.java:67]: this replacement does not return the month. It also does not explain the sad peculiarities of the month indexing.

File: src/freenet/support/CurrentTimeUTC.java [hunk 3]
```diff
@@ -29,30 +40,45 @@ public static Date get() {
...
 		synchronized(mCalendar) {
 			mCalendar.setTimeInMillis(System.currentTimeMillis());
 			return mCalendar.get(Calendar.MONTH);
 		}
 	}
 
+	/**
+	 * Get the current day of the month.
+	 * @deprecated use {@link java.time.LocalDate#now(java.time.ZoneId) LocalDate.now(ZoneOffset.UTC)}
+	 */
```
Comment (ArneBab) [inline src/freenet/support/CurrentTimeUTC.java:80]: this replacement does not return the day.

**Comment by ArneBab** [comment]
Thank you for stepping up. Currently the comments do not enable quick replacement.

Please try to put yourself in the shoes of someone who wrote a plugin three years ago and since then has only programmed websites with vue or similar. The comment needs to enable them to adjust the code without investigating, otherwise it will never be possible to turn this into deprecation for removal. There are already modified plugins in use. If it’s not trivial (in the strictest sense) to replace the deprecation, removing the deprecated version will still break plugins in the next decade.

**Comment by bertm** [comment]
I would not expect plugin developers to need this level of hand-holding in finding alternatives for these (until now mostly undocumented) methods. I've now updated the deprecation notices with verbatim copies of the new implementations provided in #866 to allow for a copy-paste migration path.

**Comment by bertm** [comment]
Updated the commits to have GPG signatures.

**Review by Bombe — APPROVED** [review]
🔥🔥🔥

**Comment by ArneBab** [comment]
@bertm merged — thank you! And welcome back! It’s great to see you again :heart:

**Pull request merged**